homeassistant/auth/mfa_modules/insecure_example.py

```diff
@@ -0,0 +1,85 @@
+"""Example auth module."""
+import logging
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from . import MultiFactorAuthModule, MULTI_FACTOR_AUTH_MODULES, \
+    MULTI_FACTOR_AUTH_MODULE_SCHEMA
+
+CONFIG_SCHEMA = MULTI_FACTOR_AUTH_MODULE_SCHEMA.extend({
+    vol.Required('users'): [vol.Schema({
+        vol.Required('user_id'): str,
+        vol.Required('pin'): str,
+    })]
+}, extra=vol.PREVENT_EXTRA)
+
+STORAGE_VERSION = 1
+STORAGE_KEY = 'auth_module.insecure_example'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@MULTI_FACTOR_AUTH_MODULES.register('insecure_example')
+class InsecureExampleModule(MultiFactorAuthModule):
+    """Example auth module validate pin."""
+
+    DEFAULT_TITLE = 'Insecure Personal Identify Number'
+
+    def __init__(self, hass, config):
+        """Initialize the user data store."""
+        super().__init__(hass, config)
+        self._data = None
+        self._users = config.get('users', [])
+
+    @property
+    def input_schema(self):
+        """Input schema."""
+        schema = OrderedDict()
+        schema['pin'] = str
+        return schema
+
+    @property
+    def users(self):
+        """Return users."""
+        return self._users
+
+    async def async_setup_user(self, user_id, data=None):
+        """Setup auth module for user."""
+        if not data:
+            raise ValueError('Expect data parameter')
+
+        pin = data.get('pin')
```

Comment:
This can't happen if part of input schema?

Reply:
No, it is not, this is `setup_schema`, may not same as `input_schema` in general.